gateway-ha/src/main/java/io/trino/gateway/ha/router/TrinoQueryProperties.java

```diff
@@ -0,0 +1,470 @@
+                yield QualifiedName.of(defaultCatalog, defaultSchema, tableParts.getFirst());
+            case 2:
+                yield QualifiedName.of(defaultCatalog, tableParts.getFirst(), tableParts.get(1));
+            case 3:
+            default:
+                yield QualifiedName.of(tableParts.getFirst(), tableParts.get(1), tableParts.get(2));
+        };
+    }
+
+    public String getBody()
+    {
+        return body;
+    }
+
+    public String getQueryType()
+    {
+        return queryType;
+    }
+
+    @SuppressWarnings("unused")
+    public String getDefaultSchema()
+    {
+        return defaultSchema;
+    }
+
+    public Set<QualifiedName> getTables()
+    {
+        return tables;
+    }
+
+    public boolean tablesContains(String testName)
+    {
+        char dot = '.';
+        char quote = '"';
+        List<Identifier> parts = new ArrayList<>();
+        int start = 0;
+        boolean inQuotes = false;
+        boolean partQuoted = false;
+        for (int i = 0; i < testName.length(); i++) {
+            if (testName.charAt(i) == quote) {
+                if (!inQuotes) {
+                    if (i != start) {
+                        log.error("Illegal position for first quote character in table name: %s", testName);
+                        return false;
+                    }
+                    start = start + 1;
+                    partQuoted = true;
+                }
+                if (inQuotes && testName.charAt(i - 1) == '\\') {
+                    continue;
+                }
+
+                inQuotes = !inQuotes;
+                continue;
+            }
+            if (testName.charAt(i) == dot && !inQuotes) {
+                if (partQuoted) {
+                    parts.add(new Identifier(testName.substring(start, i - 1)));
+                }
+                else {
+                    parts.add(new Identifier(testName.substring(start, i)));
+                }
+                start = i + 1;
+                partQuoted = false;
+            }
+        }
+        if (partQuoted) {
+            parts.add(new Identifier(testName.substring(start, testName.length() - 1)));
+        }
+        else {
+            parts.add(new Identifier(testName.substring(start, testName.length())));
+        }
+
+        return tables.contains(QualifiedName.of(parts));
+    }
+
+    @SuppressWarnings("unused")
+    public String getDefaultCatalog()
+    {
+        return defaultCatalog;
+    }
+
+    public Set<String> getCatalogs()
+    {
+        return catalogs;
+    }
+
+    public Set<String> getSchemas()
+    {
+        return schemas;
+    }
+
+    @SuppressWarnings("unused")
+    public Set<String> getCatalogSchemas()
+    {
+        return catalogSchemas;
+    }
+
+    public boolean isQuery()
+    {
+        return isQuery;
+    }
+
+    public static class AlternateStatementRequestBodyFormat
+    {
+        // Based on https://github.com/prestodb/presto/wiki/Presto-v2-client-protocol, without session
```

Comment:
Link to trino instead?
Also, add an explanation that this is only implemented in the private fork.